### PR TITLE
Fix and add Array methods

### PR DESCRIPTION
**Pull request by davidot**
I also started implementing Array#[] with sequences but that is ruby 3.
Which version do we want to emulate?

### CODE REVIEW

File: test/natalie/array_test.rb [hunk 1]
```diff
@@ -454,19 +454,244 @@
       a = []
```
Comment (davidot) [inline test/natalie/array_test.rb:454]: This file is getting very large.
What would be a good way to split it up?

Comment (seven1m) [reply test/natalie/array_test.rb:454]: Just start putting new stuff in a separate file as you go. `test/natalie/array/whatever_test.rb`

Comment (seven1m) [reply test/natalie/array_test.rb:454]: Alternatively, if you want, you can pull spec files over from https://github.com/ruby/spec like I did with `spec/core/enumerable/*`. But I know writing the tests yourself is a bit more fun in some cases, so whichever you prefer!

**Comment by seven1m** [comment]
I didn't even know about the Ruby 3 taking an Enumerator (docs say "sequence"?) as an argument to `Array#[]`. That's cool!

We can implement that, for sure. Now that Enumerators are kinda working. :wink:

File: test/natalie/array_test.rb (hunk 1 above)
Comment (davidot) [reply test/natalie/array_test.rb:454]: I probably will at least also pull in their tests since we are missing frozen tests and other edge cases
Is there some way to run all the tests in a directory?

Comment (seven1m) [reply test/natalie/array_test.rb:454]: > Is there some way to run all the tests in a directory?

No. :disappointed: I think our spec runner would need to be taught how to run more than one file.
